ua/org.eclipse.help.webapp/m/index.js

```diff
@@ -344,6 +344,12 @@
                             });
                         } else {
                             var deleteButton = createButton(scopesPage, '<%js:scopes_scope_delete_button_label%>', '<%js:scopes_scope_delete_button_description%>', function() {
+
+                                // if the scope to be deleted is set, unset it first
+                                if (searchScope.l == 4 && searchScope.t == scopeNr) {
```

Comment:
Please first make the code more clear (e.g. replace magic "4" I see here and above with proper constants) and add some documentation about `searchScope .l, .t, .s` variables meaning.

**After that** please commit the change that fixes the problem.

Reply:
Thanks for reviewing. You mean in a separate pull request, right?

Reply:
As you like. I would just insert "refactor" commit *before* the fix commit and force push two commits on same branch.

Reply:
I did it in a separate pull request: https://github.com/eclipse-platform/eclipse.platform/pull/738

I will update this pull request when it get applied.

Reply:
@howlger :  #738 is merged. Could we continue here?

Reply:
Thx for merging https://github.com/eclipse-platform/eclipse.platform/pull/738.
Resolved as requested: speaking variable names and constants instead of magic numbers.